frontend/ui/widget/keyboardlayoutdialog.lua

```diff
@@ -58,15 +59,26 @@ function KeyboardLayoutDialog:init()
 
     local buttons = {}
     local radio_buttons = {}
+
+    local keyboard_layouts = G_reader_settings:readSetting("keyboard_layouts") or {}
+    self.keyboard_state.force_current_layout = true
     for k, _ in FFIUtil.orderedPairs(self.parent.keyboard.lang_to_keyboard_layout) do
+        local text = Language:getLanguageName(k) .. "  (" .. string.sub(k, 1, 2) .. ")"
```

Comment:
bidi takes care of all this properly?

Reply:
Looks like it does it well:
![image](https://user-images.githubusercontent.com/24273478/131245976-d84feb69-8bec-42ac-ac7f-0eabcb1054b4.png)

Unless RTL readers would expect the `(ab)` on the left of LTR language names - I have really no idea:
![image](https://user-images.githubusercontent.com/24273478/131246124-19a4c15b-0b82-4cb7-978f-981151cd52b9.png)

Pinging @yparitcher @WaseemAlkurdi @zeyadas just to know how this would be expected.

It this should be changed, might be just a matter of wrapping some of these bits in a BD.wrap() like this to get the above screenshot (should look at how this kind of thing is more properly done in other places):
`local text = Language:getLanguageName(k) .. " " .. BD.wrap("(" .. string.sub(k, 1, 2) .. ")")`

(Shouldn't block merging this.)

Reply:
@poire-z Language names being to the left/right of the name of the language are a "modern" thing, therefore, there's no rule for it being to the left or the right. To the left would look better IMHO, but no problem either way.